pyteal/ast/abi/address.py

```diff
@@ -58,18 +59,14 @@ def get(self) -> Expr:
         """
         return self.stored_value.load()
 
-    def set(
+    def set(  # type: ignore[override]
```

Comment:
I'm not sure which is preferable - What do folks think?

Options:
* Keep the PR as is with mypy ignore.
* Re-add removed type parameters with usage of the to-be-added `cast` method.

During our live discussion yesterday, I didn't realize the type error will happen.  It feels to me like we should _not_ ignore the error.  

Another alternative (that's more expansive in scope) is to rework `Address` so that it extends `TypeSpec` and uses composition for value storage/retrieval.